packages/block-library/src/navigation-link/edit.js

```diff
@@ -48,7 +48,7 @@ import { store as coreStore } from '@wordpress/core-data';
 import { ItemSubmenuIcon } from './icons';
 import { name } from './block.json';
 
-const ALLOWED_BLOCKS = [ 'core/navigation-link', 'core/spacer' ];
+const ALLOWED_BLOCKS = [ 'core/navigation-link' ];
```

Comment:
Did we mean to remove the spacer from being insertable from submenus? We could maybe leave it for a follow up, it might be possible to do something like modify padding on the items instead of inserting an element for the published view.

<img width="1395" alt="Screen Shot 2021-06-25 at 10 46 07 AM" src="https://user-images.githubusercontent.com/1270189/123466112-cd8eb780-d5a3-11eb-8e0b-0a4399fb7ed1.png">

As a note for the other reviewers, existing menus with a spacer still appear, but new menus won't allow spacers to be inserted in sub-menus.

Reply:
I removed it temporarily, as there are a couple of things we need to fix for it to work properly in submenus:
* the markup issue (ideally by getting the spacer to not render markup at all, or else by wrapping it in an `li`)
* making the spacer vertical within the submenu of a horizontal nav (right now it only works vertically within vertical nav blocks)

Reply:
Update: I created #30590 to discuss the best way forward for the Spacer markup issue, and there is #30590 for its behaviour in horizontal Nav submenus. I think we can fix these separately from this PR.

Reply:
At some "phase 2" of the block, we'll want to tackle mega-menus, where the contents of a dropdown are virtually as flexible as what you can insert in a group block, so it's something we should come back to. But definitely fine to wrap a solid V1 first.